mongo/src/main/java/com/redhat/lightblue/mongo/crud/IterateAndUpdate.java

```diff
@@ -123,6 +124,8 @@ public void update(CRUDOperationContext ctx,
             measure.begin("iteration");
             int batchStartIndex=0; // docUpdateAttempts[batchStartIndex] is the first doc in this batch
             while (cursor.hasNext()) {
+                // Read from primary for read-for-update operations
+                cursor.setReadPreference(ReadPreference.primary());
```

Comment:
fixed.

On Wed, Sep 14, 2016 at 11:55 AM, Alec Henninger <notifications@github.com>
wrote:

> In mongo/src/main/java/com/redhat/lightblue/mongo/crud/
> IterateAndUpdate.java
> <https://github.com/lightblue-platform/lightblue-mongo/pull/307#discussion_r78800986>
> :
>
> > @@ -123,6 +124,8 @@ public void update(CRUDOperationContext ctx,
> >              measure.begin("iteration");
> >              int batchStartIndex=0; // docUpdateAttempts[batchStartIndex] is the first doc in this batch
> >              while (cursor.hasNext()) {
> > +                // Read from primary for read-for-update operations
> > +                cursor.setReadPreference(ReadPreference.primary());
>
> I think this needs to be set before the call to hasNext -- at that point
> we have already executed the query right?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/lightblue-platform/lightblue-mongo/pull/307/files/2f53f146687959934772235e1d50f6942acfc5a2#r78800986>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADgDDYTZRuuNQCoDVDcBCSaKZfVLyUICks5qqDUSgaJpZM4J8X7C>
> .
>